packages/arb-token-bridge-ui/src/hooks/useNetworks.ts

```diff
@@ -80,42 +100,71 @@ export function sanitizeQueryParams({
   sourceChainId: ChainId
   destinationChainId: ChainId
 } {
-  // when both `from` and `to` are undefined, default to Ethereum and Arbitrum One
+  const defaultState = {
+    sourceChainId: ChainId.Ethereum,
+    destinationChainId: ChainId.ArbitrumOne
+  }
+  // when both `sourceChain` and `destinationChain` are undefined, default to Ethereum and Arbitrum One
   if (
     typeof sourceChainId === 'undefined' &&
     typeof destinationChainId === 'undefined'
   ) {
-    return {
-      sourceChainId: ChainId.Ethereum,
-      destinationChainId: ChainId.ArbitrumOne
-    }
+    return defaultState
   }
 
-  // only `from` is undefined
+  // only `destinationChainId` is defined
   if (
     typeof sourceChainId === 'undefined' &&
     typeof destinationChainId !== 'undefined'
   ) {
-    // get the counter
+    if (!isSupportedChainId(destinationChainId)) {
+      return defaultState
+    }
+
     const [defaultSourceChainId] =
       getPartnerChainsQueryParams(destinationChainId)
     return { sourceChainId: defaultSourceChainId!, destinationChainId }
   }
 
-  // only `to` is undefined
+  // only `sourceChainId` is defined
   if (
     typeof sourceChainId !== 'undefined' &&
     typeof destinationChainId === 'undefined'
   ) {
+    if (!isSupportedChainId(sourceChainId)) {
+      return defaultState
+    }
+
     const [defaultDestinationChainId] =
       getPartnerChainsQueryParams(sourceChainId)
-    return { sourceChainId, destinationChainId: defaultDestinationChainId! }
+    return {
+      sourceChainId: sourceChainId,
+      destinationChainId: defaultDestinationChainId!
```

Comment:
can we use typescript to make sure that the return type from `getPartnerChainsQueryParams` is a non-empty array? then we don't have to use `!`

Reply:
Discussed offline, we now handle potential error from `getPartnerChainsForChainId`, and default returned array to `[]`